File: DAY2/fuelOfFuel.js
const fs = require('fs');

const calculateFuelOfFuel = (fuel,fuelOfFuel) => {
  if(fuel <=6) {
    return fuelOfFuel;
  }
   const currentFuel = Math.floor(fuel/3)- 2;
   fuelOfFuel += currentFuel;
  return calculateFuelOfFuel(currentFuel,fuelOfFuel);
}

const calculateFuel = content => {
  const masses = content.split('\n');
  let totalFuel = 0;
  masses.forEach(mass => {
      totalFuel += calculateFuelOfFuel(mass,0);
  })
  return totalFuel;
}

const main = () => {
  const content = fs.readFileSync('./DATA/mass.txt', 'utf8');
  console.log(calculateFuel(content));
};

main();

File: PY/DAY1/fuel_of_fuel.py
import math

def calc_fuel_of_fuel(fuel,fuel_of_fuel):
  if fuel <= 6:
    return fuel_of_fuel
  current_fuel = math.floor(fuel/3)-2
  fuel_of_fuel += current_fuel
  return calc_fuel_of_fuel(current_fuel,fuel_of_fuel)

def calc_fuel(content):
  total_fuel = 0
  mass = content.splitlines()
  for x in mass:
    total_fuel += calc_fuel_of_fuel(float(x),0)
  return total_fuel

f = open('mass.txt')
print(calc_fuel(f.read()))
f.close()